### PR TITLE
[Docs] Fix broken Tune links to overview and intergration

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
Signed-off-by: Artur Niederfahrenhorst <artur@anyscale.com>

## Why are these changes needed?

Screenshot 2023-02-10 at 14 10 39
Screenshot 2023-02-10 at 14 11 27

@justinvyu Heads-up: I believe this was introduced here: https://github.com/ray-project/ray/commit/613f4b006f30cf28840f0b1a88157c89b90902b0

### CODE REVIEW

File: doc/source/ray-contribute/docs.ipynb [hunk 1]
```diff
@@ -148,7 +148,7 @@
     "```\n",
     "\n",
     "The above snippet was taken from the\n",
-    "[Tune API documentation](https://github.com/ray-project/ray/blob/master/doc/source/tune/api_docs/integration.rst),\n",
+    "[Tune API documentation](https://github.com/ray-project/ray/blob/master/doc/source/tune/api/integration.rst),\n",
```
Comment (justinvyu) [inline doc/source/ray-contribute/docs.ipynb:151]: This is no longer the case -- we can remove this line. I'll create an issue about adding the new way of adding API references using `autosummary`.

**Review by justinvyu — COMMENTED** [review]
Thanks!

File: doc/source/ray-contribute/docs.ipynb (hunk 1 above)
Comment (ArturNiederfahrenhorst) [reply doc/source/ray-contribute/docs.ipynb:151]: What is no longer the case? Feel free to edit this PR if you think there is a superior fix! 🙂

Comment (justinvyu) [reply doc/source/ray-contribute/docs.ipynb:151]: The snippet it's mentioning is outdated now and doesn't match the github link anymore. We can just remove this sentence for now.

**Review by justinvyu — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]

**Comment by ArturNiederfahrenhorst** [comment]
Thanks

**Pull request merged**